universal-login-sdk/lib/api/sdk.ts

```diff
@@ -95,6 +100,14 @@ class UniversalLoginSDK {
     this.deploymentObserver = this.deploymentObserver || new DeploymentObserver(this.blockchainService, this.config!.contractWhiteList);
   }
 
+  async getBalanceObserver(ensName: string) {
+    const walletContractAddress = await this.getWalletContractAddress(ensName);
+    ensureNotNull(walletContractAddress, WalletContractNotDeployed);
+    ensureNotNull(this.config, MissingConfiguration);
+
+    this.balanceObserver = this.balanceObserver || new BalanceObserver(this.balanceChecker, walletContractAddress, ACTIVE_TOKENS);
+  }
```

Comment:
If balanceObserver is already created we don't need to check walletContractAddress or config.

```ts
    if(this.balanceObserver) {
        return this.balanceObserver 
    } else {
        const walletContractAddress = await this.getWalletContractAddress(ensName);
        ensureNotNull(walletContractAddress, WalletContractNotDeployed);
        ensureNotNull(this.config, MissingConfiguration);
        this.balanceObserver =  new BalanceObserver(this.balanceChecker, walletContractAddress, ACTIVE_TOKENS);
        return this.balanceObserver;
    }
```